Improves Boxstation medbay

St0rmC4st3r opened the pull request:
## About The Pull Request

This PR remaps boxstation medbay to be more efficient. 
Morgue moved into the place of the janitor's closet as well as expanded to hold 2 autopsy rooms:
![image](https://user-images.githubusercontent.com/34888552/89437150-dbe7a100-d74f-11ea-8ff7-38c76c239caa.png)

Janitor's closet is now in the place of the auxiliary tool storage.
![image](https://user-images.githubusercontent.com/34888552/89437225-f883d900-d74f-11ea-849b-9fbba88780db.png)

Auxiliary tool storage is moved closer to arrivals in the place of maintenance access.
![image](https://user-images.githubusercontent.com/34888552/89437335-19e4c500-d750-11ea-875b-94f70c99d833.png)

AI sat moved closer to engineering and viro moved down to get more space, as well as repiped by @Mick1299 according to new layered piping convention.
![image](https://user-images.githubusercontent.com/34888552/89437459-4993cd00-d750-11ea-9f29-57238ba95797.png)

More surgery tables for the people!
![image](https://user-images.githubusercontent.com/34888552/89437486-52849e80-d750-11ea-83a4-1b35344d993c.png)

A separate treatment centre where tiders can get their first aid and be led deeper into the medbay if they need advanced treatment. No more dosens of assistants wandering around for absolutely no reason.
![image](https://user-images.githubusercontent.com/34888552/89437531-5fa18d80-d750-11ea-9f7a-82acda5e9c65.png)

![image](https://user-images.githubusercontent.com/34888552/89437559-692af580-d750-11ea-8558-8269ee47679f.png)

Chemistry now have their own plumbing room, to which mortal doctors are actually allowed. Unfortunately, the traditional chem lab is now called Apothecary. The plumbing room also has 3 different exit ports in the new chemistry access hall.
![image](https://user-images.githubusercontent.com/34888552/89437737-9e374800-d750-11ea-8d3a-ad775af79fb1.png)

Slight remap of the cloning and genetics lab.
![image](https://user-images.githubusercontent.com/34888552/89437753-a55e5600-d750-11ea-9cd2-4ac7760efe18.png)

Cryolab is no longer awkwardly stuck in the middle of medbay.
![image](https://user-images.githubusercontent.com/34888552/89437770-abeccd80-d750-11ea-8615-c7faed569851.png)

UPD: 05.08.2020
Updated images. Requests from Bacon and Zesko processed.
![image](https://user-images.githubusercontent.com/34888552/89438659-c2dfef80-d751-11ea-8441-e7237bd7e8c2.png)

UPD: 10.08.2020
Removed Formaldehyde bottles from tables. Bottom treatment/recovery room now has a locker with dangerous, but useful chemicals.
Removed some camera blindspots. 
Added several boxes with pill bottles and sprayers to the plumbing access room.

## Why It's Good For The Game

Oh boy....
According to SS13 mapping paradigm: "Maps must be suboptimal.", but I cannot look at the horrors that current medbay generates. It is a fucking slaughterhouse! 
This is where all the assistants get when they have a slight bruse:
https://media.discordapp.net/attachments/590714573611597824/735184798842355773/unknown.png
Whenever someone competent starts plumbing medbay also embraces chaos and a fuckton of pipes:
![image](https://user-images.githubusercontent.com/34888552/88120644-14be3c80-cbcc-11ea-8faf-eeb14a4aa444.png)

And you know what is the current average round like? 2 doctors, 2 paramedics, 1 chemist and 1 CMO in medbay (not bad). CMO and a doctor are trying to fix a janitor who died for whatever reason and has his heart spoiled. After janitor dies after the first defibrillation a passerby chemist says "no use" and tries to drag the janitor to cloning which he failed to fuel with synthflesh anyway. He could snatch the patient since the stasis bed is basically in an open hallway. While CMO, Dr. One, and chemist fight and argue about the janitor for 20 minutes, Dr. Two comes dragging husked Paramedic 1 and goes to surgery, where Paramedic 2 is augmenting a miner. While entering the medbay, Dr. Two lets in 5 tiders that just got a moderate bruise after a bar fight. In 5 minutes the storage lacks half the first aid kits and the other half is on the floor empty. The ore silo now has no metal, but 15 assistants are roaming the halls with circular saws. CMO finally demoted the chemist and revives the janitor, who calmly thanks for everything and leaves. There are at least 15 corpses in various parts of the medbay and no one has any idea which are being processed and which have just arrived. The emergency shuttle has been called.


## Changelog
:cl:
del: A clearly illegal slaughterhouse to the west of the research division on boxstation has been permanently removed.
add: An actual medbay with treatment separation and more than one surgery room has been installed in the place of the removed slaughterhouse.
/:cl:

St0rmC4st3r committed:
Improves Boxstation medbay

St0rmC4st3r committed:
Improves Boxstation medbay

St0rmC4st3r committed:
Fixed sec office not having a scrubber.

St0rmC4st3r committed:
Fixed sec office not having a scrubber.

St0rmC4st3r committed:
Changes cryocell position since they were not designed to be passed behind of.

St0rmC4st3r committed:
Changes cryocell position since they were not designed to be passed behind of.

PowerfulBacon commented:
More on the actual design, but the autopsy rooms require patients to travel through the morgue or maints to access which wouldn't make sense in a realistic environemnt (even if they are never actually used in game)

PowerfulBacon commented:
That plumbing room is disgustingly big and is way too big for a completely empty room

PowerfulBacon commented:
This completely removes the secondary tool storage, source of greytiding materials?

PowerfulBacon commented:
Could we get an entire picture of what it all looks like together?

St0rmC4st3r commented:
Putting morgue closer to medbay was Firefox's input. Originally autopsy was meant to be closer. 
Plumbing room? Yes, it's big. As intended.
Secondary tool storage: Oooops! Forgot about that one. It was late. I'll stick it somewhere else today.
Yes, when I get to my PC I'll screenshot the entire medbay.

ghost commented:
do we even have autopsy as a mechanic?

St0rmC4st3r commented:
> do we even have autopsy as a mechanic?

Hmmmm. Does harvesting organs count? Also there is a surgery called experimental dissection. 
The main purpose of those two rooms is to let lone surgeons do something without everyone else running around. Like revival, brain extraction or brainwashing.

St0rmC4st3r commented:
![image](https://user-images.githubusercontent.com/34888552/88170327-2af8e680-cc26-11ea-98b2-4143987c0685.png)
Okay, autopsy is back to what it was originally intended to look like.

St0rmC4st3r commented:
![image](https://user-images.githubusercontent.com/34888552/88171330-d191b700-cc27-11ea-9895-185ab444e1e8.png)
Tool storage moved.

St0rmC4st3r commented:
![image](https://user-images.githubusercontent.com/34888552/88171410-f0904900-cc27-11ea-8418-b2dd7c8e9ab4.png)
The entire thing.

St0rmC4st3r committed:
Requested changes

St0rmC4st3r committed:
Requested changes

ghost commented:
> > do we even have autopsy as a mechanic?
> 
> Hmmmm. Does harvesting organs count? Also there is a surgery called experimental dissection.
> The main purpose of those two rooms is to let lone surgeons do something without everyone else running around. Like revival, brain extraction or brainwashing.

with autopsy i meant finding out what a dead body died from, i know para and some other servers have it.

St0rmC4st3r commented:
> > > do we even have autopsy as a mechanic?
> > 
> > 
> > Hmmmm. Does harvesting organs count? Also there is a surgery called experimental dissection.
> > The main purpose of those two rooms is to let lone surgeons do something without everyone else running around. Like revival, brain extraction or brainwashing.
> 
> with autopsy i meant finding out what a dead body died from, i know para and some other servers have it.

Wow, uh. I actually like it. But uh, `when you code it`?

PowerfulBacon commented:
This is going to be very, very opinionated and I think it should be known that I really don't like changes so anything major like this I am going in with a bias.
I think that the plumbing room looks disgusting (I've seen it on TG and it looks like trash) and we shouldn't have massive empty rooms with literally nothing in them of that size. It would be much suited being something pointless or being way smaller since it will not even be used every round and looks terrible.
After looking at it being moved, I do think that the morgue is too far away but having it swapped with the inspection rooms wouldn't make sense unless the inspection rooms were abandoned in some way. The room to the top left of the medical area looks a little bland too, but it might look better in game.
Overall in general, everything is just further apart and looks like it will be more annoying to go and get, since the distance between the entrance and storage has increased, sleepers and cruo is further away from the doors. (I don't actually see any other sleepers apart from the one up north).
I like the genetics changes, looks less annoying to use.
Chemistry south fridge just connects to their giant empty room and will never actually be used.

The changes look good and surgery and cloning look better, but my main issue is with the massive ugly and empty room dedicated to plumbing.

Sinmero commented:
Finally a plumbing room! Maybe put some basic plumbing setup there so it doesnt look so empty? I dont see chem locker (with pill bottles and stuff). Also chemists need a wrench for plumbing. Bottom chem fridge is connected with plumbing room which doesnt make sense. 
![88120251-2d7a2280-cbcb-11ea-825b-5f88a6b1e53c3](https://user-images.githubusercontent.com/38570414/88272056-c9e21900-cce0-11ea-85da-e9dcbbc67873.png)

Wont SM delaming wreck the sat now?

Sinmero commented:
There looks much better now.
![t5456uy](https://user-images.githubusercontent.com/38570414/88279280-7970b880-ccec-11ea-9409-f2dba462df67.png)

PowerfulBacon commented:
> 
> 
> There looks much better now.
> ![t5456uy](https://user-images.githubusercontent.com/38570414/88279280-7970b880-ccec-11ea-9409-f2dba462df67.png)

Most rounds it's going to have barely any machines. Went on TG with that room and it looks absolutely disgusting in all honesty, even with machines. It does not fit the theme of the station at all which is usually highly detailed and has lots of different things around, like folders on tables. The rule for mapping is that large empty areas should be avoided or used as maint area.

Sinmero commented:
> > There looks much better now.
> > ![t5456uy](https://user-images.githubusercontent.com/38570414/88279280-7970b880-ccec-11ea-9409-f2dba462df67.png)
> 
> Most rounds it's going to have barely any machines. Went on TG with that room and it looks absolutely disgusting in all honesty, even with machines. It does not fit the theme of the station at all which is usually highly detailed and has lots of different things around, like folders on tables. The rule for mapping is that large empty areas should be avoided or used as maint area.

No i meant that there should be some default plumbing setup at roundstart. 
I think that atmos is a good example - it takes a lot of space and has a roundstart setup ready. There is plenty of room for atmosians to improve it but it works.
Plumbing machines dont really fit the station aesthetics and i blame the sprites. We need better ones.

PowerfulBacon commented:
Something like this, or having the room to the right as a maint room could work better
![image](https://user-images.githubusercontent.com/26465327/88320299-4bec3500-cd15-11ea-9e92-a8286e1a7f2e.png)

Just putting something pointless there that can be deconstructed, such as
 - Testing lab with reinforced floor area like in xeno (Makes sense with chemistry)
 - Maint room
 - Area like atmos with wall in middle
 - A few tables and chairs like an eating area

With the breakroom to the left
(Also you removed the break room which needs to go somewhere)

zeskorion reviewed (changes requested):
-you have replaced all of the random maint plating with turf passthrough. please undo this
-undocumented change to maint near locker rooms
-the one firecloset in the box hallways has been removed. care to add a few fire lockers around? box has way too few of them anyways
-as bacon has stated, you need to fill the negative space in the plumbing room. perhaps reinforce walls around chem/"apothecary" while you're at it. (you could fill it with a number of things: breakroom, test chamber, monky pen, random clutter... maybe even use random map templates?)
-you've removed a medivend. Perhaps take viro's personal medivend and put it in medbay proper, especially in the apothecary
-viro atmos is removed. Where did you put it?

Mick1299 committed:
Updates ai sat piping, fixes two bad layer manifolds

Mick1299 committed:
Updates ai sat piping, fixes two bad layer manifolds

St0rmC4st3r committed:
Merge pull request #2 from Mick1299/fdmm-patch-42022495-Mick1299

Updates ai sat piping, fixes two bad layer manifolds

St0rmC4st3r commented:
Converted to draft, since it still has plenty of stuff to fix.

St0rmC4st3r committed:
Updates almost everything

St0rmC4st3r committed:
Updates almost everything

St0rmC4st3r committed:
Oops

St0rmC4st3r committed:
Oops

Sinmero commented:
AI has lots of blind spots.

![1](https://user-images.githubusercontent.com/38570414/89517643-ab047c00-d7e2-11ea-835c-dd9935e9e560.png)
![2](https://user-images.githubusercontent.com/38570414/89517649-ae980300-d7e2-11ea-8c25-fbcc3327d27b.png)
![3](https://user-images.githubusercontent.com/38570414/89517657-b192f380-d7e2-11ea-8c90-4d9acf19387f.png)

3 formaldehyde bottles right in the open (two at chem entrance and one at the bottom treatment room). It is pretty dangerous by itself and someone can make heparin without chem heater with it.
![4](https://user-images.githubusercontent.com/38570414/89518422-8ceb4b80-d7e3-11ea-899d-8a487b9e41b6.png)

Security office has no camera monitor.

Atmos apc power is draining really fast roundstart for some reason.

5 medical vendors? Two treatment rooms with a sleeper and two stasis machines at each room? Also there are two wallmeds at both surgery room and north treatment rooms.  I think its just an extreme overabundance of med supplies and equipment. Box is a low-med pop map. Even Delta has a single sleeper and two medvendors.

![5](https://user-images.githubusercontent.com/38570414/89520729-fa4cab80-d7e6-11ea-90fc-da37ff518611.png)

I think its a good idea to change bottom treatment room into a break room.

St0rmC4st3r commented:
Alright, sticking cameras around. Probably will remove 2 vendors. 
>Formaldehyde. 

You know formaldehide is used for organ preservation in corpses, right? Since you are concerned, I can make a secure medical closet that contains Medical supplies that might be needed in emergency rooms. It will also take the role of one of the medical vendors, but more specialized. 

As for the point that Delta has less resources, I shall answer that it is also a slaughterhouse, but to a slightly lower extent and I plan to rework it too, as well as Meta and Pubby.

St0rmC4st3r committed:
Adds a medical closed with potentially dangerous drugs

St0rmC4st3r committed:
Adds a medical closed with potentially dangerous drugs

St0rmC4st3r committed:
Fixes conflict

This reverts commit b12e82fcf256622672a9a1a316e5f876f9b94c91.

St0rmC4st3r committed:
Removes Calomel syringe duplicate

St0rmC4st3r committed:
Removes Calomel syringe duplicate

St0rmC4st3r committed:
Fixes conflict

This reverts commit b12e82fcf256622672a9a1a316e5f876f9b94c91.

St0rmC4st3r committed:
Removes formaldehyde bottles in the open, replaced camera, added closet

St0rmC4st3r committed:
Removes formaldehyde bottles in the open, replaced camera, added closet

St0rmC4st3r commented:
@zeskorion Any other requests? If not, can you put it to vote?

St0rmC4st3r committed:
Fixes a few issues that came up during testmerge.

St0rmC4st3r committed:
Fixes a few issues that came up during testmerge.

St0rmC4st3r committed:
Fix conflicts 1?

St0rmC4st3r committed:
Fix conflicts 1?

St0rmC4st3r committed:
Fix Conflicts 2?

St0rmC4st3r committed:
Fix Conflicts 2?

St0rmC4st3r commented:
So. In the last commit some stuff changed after the testmerge:
Western doors removed due to defeating the design intent of the entire PR which is warding tiders away from operating areas.
Added a maint door to the med storage. 
Fixed access on some maintenance doors so that you need med access to get into the medbay.
Added some chairs to chemistry.
Little rec area for doctors.
You can leave the restricted medbay are on your own.
Other minor fixes.

PowerfulBacon commented on code/game/objects/structures/crates_lockers/closets/secure/medical.dm:
```diff
@@ -50,6 +50,28 @@
+	new /obj/item/reagent_containers/syringe/diphenhydramine(src)
+	new /obj/item/reagent_containers/hypospray/medipen/atropine(src) //just one
+	new /obj/item/storage/box/syringes(src)
+	return
```
Pointless return

PowerfulBacon commented:
![image](https://user-images.githubusercontent.com/26465327/92325351-b1844e80-f041-11ea-805a-8059dd5fb7ac.png)
these decals

PowerfulBacon commented:
Also personally I think the rooms next to morgue that will never be used should be removed, morgue should be shifted over and jani closet should be fit back where it was

PowerfulBacon commented:
Chem room should have a toolbox for removing the middle section

PowerfulBacon commented:
Also chemistry has blast doors rather than shutters which seems a little overkill

PowerfulBacon commented:
Untestmerged

4rchibald commented:
@St0rmC4st3r and also @PowerfulBacon @Sinmero since you seem to care enough about this PR 

As it was already pointed out it really seem to be placing things too far apart. I know you claim its a design, its one thing to make area not crammed and another to go out of your way to make it completely impractical and honestly tiresome to work around. 
Morgue should be closer to cloning lab, with current location bodies will never be moved to morgue and there's literally zero chance that anyone will ever check on it to see if body has still a soul (chances for that are small already, making them even lower is not a good idea). 
I'm proposing bringing jani closet back to its original place (as engineer I must say I also missed that shortcut from atmos straight to central hallway. Doesn't seem like much but it really makes things smoother, especially in emergencies). Morgue should be moved to the location of medical storage alongside with autopsy room. 
Medical storage could be moved below chemlab, to first make your plumbing room smaller and to move meds 'closer 
to the action' (even then I'm afraid it may be too far from 'entrance area' where most of the healing will occur). It also makes sense that chemlab would have window and fridge to supply doctors straight to the storage. 

I marked entrance area on green. Not exactly sure what to do with it, maybe add walls, windows an firelocks to break that room a little bit, I'm concerned about 3 things. 
First, chaos-it seems like people will get really confused, everyone will be coming left and right, some coming in, some coming out, some going for supplies, fresh bodies will get mixed with soulless and too damaged to immediately deal with, there's not even clear side corner to push catatonic into. Another thing is atmos, especially now with fastmos its all a pain in ass so I would really like to avoid bigger room with no firelocks. And lastly, of course, looks-its almost as plain and ugly as plumbing room, just big rectangle, adding some walls/windows maybe alongside green lines are added with that wall below mover one tile to the right would make that space look and work better-there would be clear passage and clear area to work on patients.  
I also feel obligated to say I liked what you did with curtains in bottom though, that's a nice implementation of them.

Plumbing room even after making it smaller needs something in it so its not just a big empty rectangle in the middle of station. Either add those lockers and wall in middle like others suggested or maybe mask it as something else-add some beds etc to make it look like medical area. Beds are easy to deconstruct and I even suspect they don't collide with plumbing anyway so we would all know that area is intended for plumbing but when its not used for that purpose it still looks legit.
![proposed changes](https://user-images.githubusercontent.com/54917446/92390182-aea95c80-f11a-11ea-97f6-6764c2e05170.png)

jlsnow301 commented:
Closing #2422 since this is massive and I want to just make my PR off of this instead

St0rmC4st3r committed:
Final touches.

St0rmC4st3r commented:
@4rchibald thanks for the feedback. 
Although you are using the older version of the map, you have multiple good points and some issues with proposed changes. 
First of all, moving the medical storage in the gap between chem and the hallway makes a lot of sense, since right now that gap is used exactly for that reason but as a secondary medical storage. It is a good idea. But I don't think it is possible to do at this point due to how this secondary medbay storage is built to be a junction between the hall, chem, and plumbing lab. If you really want it - you can try it as a separate PR.
The plumbing room has been made smaller and less empty long ago.
In the entrance area the soulless bodies should not get mixed with fresh. The exam room is intended for small injuries. Both types of dead should be brought further inside.
